### ISSUE TITLE
Quoted curly braces identifier in f-string is not properly replaced

### ISSUE THREAD

**Issue by mlao-pdx** [python-rope/rope#303]
```python
# Refactor issue -> bug
issue = "Rope inserts after first character of replacement instead replacing when quoted in f-strings."

# Failing examples will look like {ibugssue}
success = f" {issue} "
bug_example_1 = f" '{issue}' "
bug_example_2 = f""" "{issue}" """
bug_example_3 = f' "{issue}" '
```

**Comment by chrisjsewell** [comment]
hey yeh I'm really surprised more people haven't commented on this.
I use f-strings all the time, and it makes rope really horrible to use in the current state

**Comment by lieryan** [comment]
@mlao-pdx Thank you for reporting this bug, this will be fixed in 0.21

cc: @all-contributors

**Comment by mlao-pdx** [comment]
Awesome, thanks @lieryan.
